Ryujinx.HLE/Input/HidProController.cs

```diff
@@ -0,0 +1,44 @@
+﻿namespace Ryujinx.HLE.Input
+{
+    public class HidProController : HidControllerBase
+    {
+        bool Wired = false;
+
+        public HidProController(Switch Device) : base(HidControllerType.ProController, Device)
+        {
+            Wired = true;
+        }
+
+        public override void Connect(HidControllerId ControllerId)
+        {
+            base.Connect(ControllerId);
+
+            HidControllerColorDesc SingleColorDesc =
+                HidControllerColorDesc.ColorDesc_ColorsNonexistent;
+
+            HidControllerColorDesc SplitColorDesc = 0;
+
+            NpadColor SingleColorBody    = NpadColor.Black;
+            NpadColor SingleColorButtons = NpadColor.Black;
+
+            Device.Memory.WriteInt32(Offset + 0x8,  (int)SingleColorDesc);
+            Device.Memory.WriteInt32(Offset + 0xc,  (int)SingleColorBody);
+            Device.Memory.WriteInt32(Offset + 0x10, (int)SingleColorButtons);
+            Device.Memory.WriteInt32(Offset + 0x14, (int)SplitColorDesc);
+
+            Connected = true;
+        }
+
+        public override void SendInput(
+            HidControllerButtons Buttons,
+            HidJoystickPosition LeftStick,
+            HidJoystickPosition RightStick)
+        {
+            long ControllerOffset = WriteInput(Buttons, LeftStick, RightStick, HidControllerLayouts.Pro_Controller);
+
+            Device.Memory.WriteInt64(ControllerOffset + 0x28,
+              Connected ? (uint)HidControllerConnState.Controller_State_Connected : 0 |
```

Comment:
You need to wrap the first ternary operation in brackets, as bitwise-OR has higher precedence in C#.

```suggestion
              (Connected ? (uint)HidControllerConnState.Controller_State_Connected : 0) |
```